src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -367,6 +367,12 @@ protected FormattableString GetResourceTypeExpression(ResourceTypeSegment resour
             {
                 return $"{candidates.First().Type}.ResourceType";
             }
+
+            if (string.IsNullOrEmpty(resourceType.SerializedType))
+            {
+                throw new InvalidOperationException($"ResourceType is empty");
```

Comment:
Regen the sdk repo

Reply:
There is no SDK change in the regen [PR](https://github.com/Azure/azure-sdk-for-net/pull/42735)